### PR TITLE
(US-76) feat: add language selector in footer

### PR DESCRIPTION
**Pull request by lucachaco**
- Language selector in UI added
- Detects default language from browser
- Remembers selections after closing modal

image

### CODE REVIEW

**Review by jessgusclark — CHANGES_REQUESTED** [review]
A few things.

the styles for the dropdown are missing:
image
image

When I choose Spanish and then click on Metamask, the "connecting" message is still in English:

Screen Shot 2021-08-12 at 1 30 42 PM

It reverts to English:
  1. Click on connect
  2. change to Spanish
  3. Click on WalletConnect
  4. Close the QR code with the (x)
  5. Click on Login. 
  6. Language is set to English but the text is in Spanish.

Can you add Cypress tests to test the correct functionality? There are a few of them there that should be a good reference.

**Comment by lucachaco** [comment]
I just thought of pushing the fixes of today so you guys can take a look of the progress so far.

- Selector border remove
- Cypress tests added
- Issue reverting back to english fixed.
Pending: Styling the opened box selector.

image
image

File: src/ux/step1/WalletProviders.tsx [hunk 1]
```diff
@@ -22,6 +25,21 @@ const ProvidersWrapper = styled.div`
   padding: 8px;
 `
 
+const LanguageSelector = styled.select`
+  float: left;
+  border: 0;
+`
```
Comment (jessgusclark) [inline src/ux/step1/WalletProviders.tsx:31]: It is missing the color, size, and removal of the outline when it is focused. Try this:

```suggestion
const LanguageSelector = styled.select`
  float: left;
  border: 0;
  color: #B0AEB1;
  size: 12px;
  &:focus {
    outline: none;
  }
`
```

File: src/ux/step1/WalletProviders.tsx [hunk 2]
```diff
@@ -22,6 +25,21 @@ const ProvidersWrapper = styled.div`
...
+
+const NoWalletFooter = styled.span`
+  float: right;
+  padding: 2px;
+  
+`
+const FooterWrapper = styled.div`
+  height: 35px;
+  padding: 0 8px 0 8px;
```
Comment (jessgusclark) [inline src/ux/step1/WalletProviders.tsx:40]: Height should rarely be used. padding would be better:

```suggestion
  padding: 0 8px 25px 8px;
```

**Review by jessgusclark — CHANGES_REQUESTED** [review]
Worked as expected and the tests were great!!

Just a few CSS changes.

**Comment by lucachaco** [comment]
Thanks for the help on the style @jessgusclark .

- Selector style done
- Cypress tests done
- Issue reverting back to english done.

image

File: src/Core.tsx [hunk 3]
```diff
@@ -341,16 +351,18 @@ export class Core extends React.Component<IModalProps, IModalState> {
       // reset state
       this.setState(INITIAL_STATE)
     }
-
+    // this fetches all available languages in this form [{en:english},...]
+    const availableLanguages = Object.entries(i18next.services.resourceStore.data).map(keyValueLanguage => { return { code: keyValueLanguage[0], name: keyValueLanguage[1].name.toString() } })
```
Comment (ilanolkies) [inline src/Core.tsx:355]: We don't need to calculate this in each `render()`. We can do it only once outside the method scope.

File: src/ux/step1/WalletProviders.tsx [hunk 4]
```diff
@@ -47,11 +69,23 @@ export const WalletProviders = ({ userProviders, setLoading }: IWalletProvidersP
...
-  <Paragraph className={PROVIDERS_FOOTER_TEXT_CLASSNAME}>
 
-    <Trans>No wallet? </Trans>
-    <NoWalletAnchor href="https://developers.rsk.co/wallet/use/" target="_blank" className={ANCHOR_CLASSNAME}>
-      <Trans>Get one here!</Trans>
-    </NoWalletAnchor>
-  </Paragraph>
+  <FooterWrapper >
+    <Paragraph>
+      <LanguageSelector onChange={(val) => changeLanguage(val.target.value)} defaultValue={selectedLanguageCode} name="languages" id="languages">
```
Comment (ilanolkies) [inline src/ux/step1/WalletProviders.tsx:75]:
```suggestion
      <LanguageSelector onChange={(e) => changeLanguage(e.target.value)} defaultValue={selectedLanguageCode} name="languages" id="languages">
```

More declarative, for`event`

**Review by ilanolkies — COMMENTED** [review]
LRGTM!

**Review by ilanolkies — APPROVED** [review]

**Review by ilanolkies — APPROVED** [review]

**Pull request merged**